### DISCUSSION TITLE
Changing the baseline airfoil : FOAM FATAL IO ERROR

### DISCUSSION THREAD

**Discussion by usmanshahid90** [mdolab/dafoam#120] [General]
I am changing the baseline airfoil for optimization from NACA0012 to NACA6 series. Saving the coordinates as per the guidance of DAFoam FAQ,  I am having this error.

FOAM FATAL IO ERROR
Attempt to get back from bad stream
file : volumemesh.xyz at line 1
From function bool foam : : Istream : : getback (Foam: :token&)
in file db/IOstreams/IOstreams/Istream.C at line 56.
FOAM exiting

Generating Mesh .. Done

Solution is running even after this error but why this error? Plus, after the solution run, when i see export the baseline coordinates, they again are of NACA0012. Not of 6 series.
Kindly guide me in this regard

**Comment by friedenhe** [comment]
The error message should be in logMeshGeneration.txt

**Reply by usmanshahid90** [reply]
![Image](https://user-images.githubusercontent.com/75417534/114379985-f6a5cb80-9ba2-11eb-9a07-8e01a585f5e9.jpeg)
 This is the logMeshGeneration.txt file. 

Kindly Guide me.

**Reply by friedenhe** [reply]
The mesh was not generated. Check your airfoil profiles and see if they are from 0 to 1 and from leading to the trailing edge. Also, the lower surface should be PS and the upper surface should be SS.

**Reply by usmanshahid90** [reply]
Kindly check that. I manipulated some trailing edge points. Now this error is being pop out.
Cannot find patchField entry for auto5.

What its doing is,in constant<polyMesh, all the files are updating with boundary file . Here auto5 is new thing.
What should I do?
Kindly guide me.
Thanks
![error](https://user-images.githubusercontent.com/75417534/114812038-a9a33e80-9dc8-11eb-9063-e46d9b4a7b78.jpeg)
![WhatsApp Image 2021-04-15 at 8 53 11 AM](https://user-images.githubusercontent.com/75417534/114812046-adcf5c00-9dc8-11eb-9878-03f2a6766ff9.jpeg)

**Reply by friedenhe** [reply]
In preProcessing.sh, change `autoPatch 30 -overwrite >> logMeshGeneration.txt` to `autoPatch 45 -overwrite >> logMeshGeneration.txt` and try again. BTW: can you upload the log file instead of screenshot?

**Reply by usmanshahid90** [reply]
I have tried this change. It is generating mesh successfully but when I run the solution, It says:
--> FOAM FATAL IO ERRROR:
Cannot find patchField entry for auto5
Kindly guide me in this regard.

Actually the PC which I am using for running analysis isnt connected to the internet and there is also some issue with its USB port. Thats why i was sending u the screenshots.

Thanks

**Reply by friedenhe** [reply]
Can you increase to `autoPatch 60 -overwrite >> logMeshGeneration.txt`? If it is still not working, upload the case here.

**Reply by usmanshahid90** [reply]
hy,
I tried it too but it didnt work. I will upload the files soon.
Before that, can u please tell me would changing the baseline airfoil would make any difference in the optimized result if constraints and objective functions are same?

Kindly guide me.

**Reply by friedenhe** [reply]
If you use a very large number of FFD points and converge the optimization tightly (like this [paper](https://www.sciencedirect.com/science/article/abs/pii/S1270963818319072)), the results will be the same. But because we usually use like 20 FFD points, the result will be slightly different.

**Reply by usmanshahid90** [reply]
Alright, just need to discuss it ; I've read in surrogate modelling research papers that gradient based methods are prone to stuck in local minima. But as per the research paper you mentioned, it seem like starting from circle and ending to airfoil, it doesnt stuck in local minima. what you'll say about it?

**Reply by friedenhe** [reply]
This is an issue only if your optimization problem has multiple local minima. We have not found a multiple local minima case for RANS-based wing/airfoil aerodynamic shape optimization. If you find one, let us know, and we will be more than happy to try it.
